src/assign.c

```diff
@@ -897,14 +897,14 @@ const char *memrecycle(const SEXP target, const SEXP where, const int start, con
       switch (TYPEOF(source)) {
       case REALSXP: if (sourceIsI64)
                     CHECK_RANGE(int64_t, REAL, val!=NA_INTEGER64 && (val<=NA_INTEGER || val>INT_MAX),   PRId64,  "out-of-range (NA)", val)
```

Comment:
```suggestion
                    CHECK_RANGE(int64_t, REAL, val!=NA_INTEGER64 && (val<=INT_MIN || val>INT_MAX),   PRId64,  "out-of-range (NA)", val)
```

It feels cleaner and we check in init.c anyway if (int)NA_INTEGER != (int)INT_MIN